src/backend/distributed/commands/utility_hook.c

```diff
@@ -1627,6 +1644,12 @@ RunPreprocessMainDBCommand(Node *parsetree, const char *queryString)
 static void
 RunPostprocessMainDBCommand(Node *parsetree)
 {
+	if (!IsStatementSupportedIn2Pc(parsetree) ||
+		!IsStatementMarkDistributedFor2PC(parsetree))
```

Comment:
Only the second check seems to be enough

Reply:
Actually, these lines don't seem to add any value for this PR, since we still have the if statement below.